brambl-sdk/src/main/scala/co/topl/brambl/syntax/TokenTypeIdentifierSyntax.scala

```diff
@@ -0,0 +1,110 @@
+        case (GROUP, qd, Some(gId), Some(sId), _, None) => GroupFungible(gId, sId.value, qd)
+        // If groupAlloy is not provided, the groupId is used to identify instead
+        case (SERIES, qd, Some(gId), Some(sId), None, _) => SeriesFungible(sId, gId.value, qd)
+        case _                                           => throw new Exception("Invalid asset")
+      }
+    case _ => throw new Exception("Invalid value type")
+  }
+}
+
+class TypeIdentifierToQuantityDescriptorSyntaxOps(val typeIdentifier: ValueTypeIdentifier) extends AnyVal {
+
+  def getQuantityDescriptor: Option[QuantityDescriptorType] = typeIdentifier match {
+    case GroupAndSeriesFungible(_, _, qd) => Some(qd)
+    case GroupFungible(_, _, qd)          => Some(qd)
+    case SeriesFungible(_, _, qd)         => Some(qd)
+    case _                                => None
+  }
+}
+
+/**
+ * Identifies the specific type of a token.
+ */
+trait ValueTypeIdentifier
+
+/**
+ * A LVL value type
+ */
+case object LvlType extends ValueTypeIdentifier
+
+/**
+ * A Group Constructor Token value type, identified by a GroupId
+ *
+ * @param groupId The GroupId of the Group Constructor Token
+ */
+case class GroupType(groupId: GroupId) extends ValueTypeIdentifier
+
+/**
+ * A Series Constructor Token value type, identified by a SeriesId
+ * @param seriesId The SeriesId of the Series Constructor Token
+ */
+case class SeriesType(seriesId: SeriesId) extends ValueTypeIdentifier
+
+sealed trait AssetType extends ValueTypeIdentifier
+
+/**
+ * A Group and Series fungible asset type, identified by a GroupId, a SeriesId, and a QuantityDescriptorType.
+ *
+ * @param groupId The GroupId of the asset
+ * @param series The SeriesId of the asset
+ * @param qdType The QuantityDescriptorType of the asset
+ */
+case class GroupAndSeriesFungible(groupId: GroupId, series: SeriesId, qdType: QuantityDescriptorType) extends AssetType
+
+/**
+ * A Group fungible asset type, identified by a GroupId, a Series alloy, and a QuantityDescriptorType. If the asset is
+ * not an alloy, the series "alloy" is given by the seriesId.
+ *
+ * @param groupId  The GroupId of the asset
+ * @param seriesAlloyOrId If the asset is an alloy, the Series alloy. Else the SeriesId of the asset
+ * @param qdType The QuantityDescriptorType of the asset
+ */
+case class GroupFungible(groupId: GroupId, seriesAlloyOrId: ByteString, qdType: QuantityDescriptorType)
+    extends AssetType
+
+/**
+ * A Series fungible asset type, identified by a SeriesId, a Group alloy, and a QuantityDescriptorType. If the asset is
+ * not an alloy, the group "alloy" is given by the groupId.
+ *
+ * @param series The SeriesId of the asset
+ * @param groupAlloyOrId If the asset is an alloy, the Group alloy. Else the GroupId of the asset
+ * @param qdType The QuantityDescriptorType of the asset
+ */
+case class SeriesFungible(series: SeriesId, groupAlloyOrId: ByteString, qdType: QuantityDescriptorType)
```

Comment:
In GroupFungible,  id was named "groupId", maybe here you can do the same?
```suggestion
case class SeriesFungible(seriesId: SeriesId, groupAlloyOrId: ByteString, qdType: QuantityDescriptorType)
```

Reply:
Good catch! I will make the changes